File: ৫২টি প্রোগ্রামিং সমস্যা ও সমাধান/প্রোগ্রামিং সমস্যা ৯ — পূর্ণবর্গ সংখ্যা.c
#include <stdio.h>
#include <math.h>

int main(){
    int T, i, N, root;
    scanf("%d", &T);
    
    for(i = 1; i <= T; i++){
        scanf("%d", &N);
        
        // If negative, it's not a perfect square.
        if(N < 0){
            printf("NO\n");
            continue;
        }
        
        double sr = sqrt(N);
        root = (int) sr;  // This is floor(sr) when sr is positive
        
        // Check both root and root+1 in case of precision issues.
        if(root * root == N || (root+1) * (root+1) == N)
            printf("YES\n");
        else
            printf("NO\n");
    }
    return 0;
}

File: ৫২টি প্রোগ্রামিং সমস্যা ও সমাধান/প্রোগ্রামিং সমস্যা ৯ — পূর্ণবর্গ সংখ্যা.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int T = scanner.nextInt();  // Read number of test cases
        
        for (int i = 0; i < T; i++) {
            int N = scanner.nextInt();  // Read the integer N
            
            // If N is negative, it's not a perfect square.
            if (N < 0) {
                System.out.println("NO");
                continue;
            }
            
            // Compute the square root as a double.
            double sr = Math.sqrt(N);
            // Cast to int, which gives the floor value for positive numbers.
            int root = (int) sr;
            
            // Check both root*root and (root+1)*(root+1) to account for precision issues.
            if (root * root == N || (root + 1) * (root + 1) == N) {
                System.out.println("YES");
            } else {
                System.out.println("NO");
            }
        }
        
        scanner.close();
    }
}
